funding/manager.go

```diff
@@ -3601,3 +3676,49 @@ func (f *Manager) deleteChannelOpeningState(chanPoint *wire.OutPoint) error {
 		return bucket.Delete(outpointBytes.Bytes())
 	}, func() {})
 }
+
+// updateShortChannelID waits for confirmation to update the real short channel
+// id for this 	not yet confirmed channel.
+func (f *Manager) updateShortChannelID(ch *channeldb.OpenChannel) (
+	*lnwire.ShortChannelID, error) {
+
+	log.Infof("waiting for confirmation to update short channel id %v", ch.ShortChannelID)
+	confirmedChan := ch
+	confChan := make(chan *confirmedChannel, 1)
+	cancelChan := make(chan struct{})
+	skipChan := make(chan lnwire.ShortChannelID)
+
+	f.wg.Add(1)
+	go f.waitForFundingConfirmation(ch, cancelChan, confChan, 1)
+	defer close(skipChan)
+	defer close(cancelChan)
+
+	select {
+	case conf, ok := <-confChan:
+		var err error
+		if !ok {
+			return nil, fmt.Errorf("failed to wait for funding confirmation")
+		}
+		confirmedChan, err = f.cfg.FindChannel(
+			lnwire.NewChanIDFromOutPoint(&ch.FundingOutpoint))
+		if err != nil {
+			return nil, fmt.Errorf("failed to find channel %v %v", ch.ShortChannelID, err)
+		}
+		err = f.cfg.Wallet.ValidateChannel(confirmedChan, conf.fundingTx)
+		if err != nil {
+			return nil, fmt.Errorf("unable to validate channel: %v", err)
+		}
+		if err := confirmedChan.MarkAsOpen(conf.shortChanID, true); err != nil {
```

Comment:
This will create another `Packager` under a different short channel id.  I think this can break the forwarding package logic if there is an existing payment that needs to be fulfilled during this upgrade

Reply:
Following @cfromknecht feedback (https://github.com/lightningnetwork/lnd/pull/4424#issuecomment-654487642) I set this PR as depends on https://github.com/lightningnetwork/lnd/pull/4459.
That PR converts the packager to use static keys that don't change upon confirmation.